### ISSUE TITLE
Use "StyleUnit" when setting properties

### ISSUE THREAD

**Issue by RoyJacobs** [bschwind/yoga-rs#2]
First of all, excellent library! I'm trying to use it instead of an alternative Yoga wrapper for which [I've sent a pull request](https://github.com/dobrite/yoga-wrapper-rs/pull/1) to use YGValues in lieu of raw floats.

I've noticed you already have the enum StyleUnit set up. For things like setting height and width I believe this enum should be used to construct a YGValue, instead of just passing in an f32. This allows users of Yoga to also use percentage sizes, instead of just absolutes.

**Comment by bschwind** [comment]
Hey great suggestion! Let me try it out in a branch and see how it goes. I actually haven't finished this library completely yet which is why it's not yet published on crates.io so just keep that in mind ;)

**Comment by RoyJacobs** [comment]
That's fine! I think the Yoga C API actually exposes a couple of automatically generated setters for these properties (e.g. `YGNodeStyleSetWidth`, `YGNodeStyleSetWidthPercent` and `YGNodeStyleSetWidthAuto`).

It should be possible to match on the incoming StyleUnit and call the appropriate method. I can give it a shot as well, if you like.

And, yes, I'm aware that it's not finished :D But it seems useable enough for me to experiment a little bit with it.

**Comment by bschwind** [comment]
Yep I just realized it generated those different setters. Working on it now, I'll comment back when I have something.

**Comment by RoyJacobs** [comment]
Awesome :yellow_heart:

**Comment by bschwind** [comment]
Should a StyleUnit of `Undefined` be a no-op?

**Comment by RoyJacobs** [comment]
I'm not sure, but it seems like you would set it to undefined in order to indicate that you don't care about that value (and that Yoga may choose to change it when calculating the layout).

**Comment by RoyJacobs** [comment]
Also, when setting the unit to auto it sets the value to undefined: https://github.com/facebook/yoga/blob/master/yoga/Yoga.c#L635

**Comment by bschwind** [comment]
I'm trying to think of the best way to represent these values in my `FlexStyle` enum. Right now I have an API that looks like this which I think is convenient:

```
root.apply_styles(vec!(
    Width(100.0),
    Height(100.0)
));
```

But if I changed the `f32`s to `YGValue`, it would be something like:

```
root.apply_styles(vec!(
    Width(YGValue::Percent(100.0)),
    Height(YGValue::Percent(100.0))
));
```

which is a little tedious. What do you think about something like this instead?

```
root.apply_styles(vec!(
    WidthPercent(100.0),
    HeightPercent(100.0)
));
```

Still not sure what I like best. Is there any way you can set the code up such that you could call it like this?

```
pub enum FlexStyle {
    Height(YogaValue),
    ...
    Width(YogaValue)
}

root.apply_styles(vec!(
    Width(100.0),
    Height(YogaValue::Percent(50.0))
));
```

and have the f32 automatically get converted into a `YogaValue`. I don't believe you can store trait objects in enums which is probably what I would need, right?

**Comment by RoyJacobs** [comment]
I think providing both `Width` and `WidthPercent` could result into ambiguities (what if you set both?).

I wouldn't mind having to use `Percent`, `Point` or `Auto` explicitly. It doesn't need to be prefixed all the time if you import it, either.

I'm not sure what would be "idomatic Rust" here, though!

**Comment by bschwind** [comment]
Yeah the ambiguity had me a bit worried too though the way my API is set up, I think it's "last one in wins"

I guess I'll go with the `Percent`, `Point`, `Auto` way for now.

**Comment by bschwind** [comment]
@RoyJacobs https://github.com/bschwind/yoga-rs/pull/3

**Comment by bschwind** [comment]
Implemented by #3
